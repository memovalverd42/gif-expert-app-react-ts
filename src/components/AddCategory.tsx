import { ChangeEvent, Dispatch, FormEvent, SetStateAction, useState, FC } from 'react';

interface AddCategoryProps {
    setCategories: (categories: string[]) => void;
    categories: string[]
}

interface newCategoryProp {
    onNewCategory: ( newCategory: string ) => void
}

// export const AddCategory: FC<AddCategoryProps> = ( { setCategories, categories } ) => {
export const AddCategory: FC<newCategoryProp> = ( { onNewCategory } ) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ( { target }: ChangeEvent<HTMLInputElement> ) => {
        setInputValue(target.value);
    }

    const onSubmit = ( event: FormEvent<HTMLFormElement> ) => {
        event.preventDefault();

        const trimedInputValue = inputValue.trim();

        if( trimedInputValue.length <= 1 ) return;

        // setCategories( [ inputValue, ...categories ] );
        onNewCategory(trimedInputValue.trim());
        setInputValue('');
    }

    return (
        <form 
            onSubmit={ onSubmit }
            aria-label='form'
        >
            <input 
                type="text" 
                placeholder="Buscar GIF"
                value={inputValue}
                onChange={ onInputChange }
            />
        </form>
    )
}
