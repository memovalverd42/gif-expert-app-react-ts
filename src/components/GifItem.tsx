import { FC } from "react"
import { Image } from "../types/image.types"

export const GifItem: FC<Image> = ( {title, url} ) => {

    return (
        <div className="card">
            <img src={ url } alt={ title }/>
            <p>{ title }</p>
        </div>
    )

}
