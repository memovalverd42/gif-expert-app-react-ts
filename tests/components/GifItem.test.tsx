import React from "react";
import { getByRole, render, screen } from "@testing-library/react";
import { GifItem } from '../../src/components/GifItem';
import { Image } from "../../src/types/image.types";


describe('Tests del componente <GifItem/>', () => {

    const spiderman: Image = {
        id: '23423',
        title: 'Spiderman',
        url: 'asdfadsfasdf'
    }
    
    test('debe de hacer match con el snapshot', () => {

        const { container } = render(
            <GifItem { ...spiderman } />
        );
        
        expect( container ).toMatchSnapshot();
        
    });
    
    
    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
        render(
            <GifItem { ...spiderman } />
        );
        
        const { src, alt } = screen.getByRole<HTMLImageElement>('img');
        // expect( src ).toBe( spiderman.url );
        expect( alt ).toBe( spiderman.title );
        
    });
    
    test('debe de mostrar el titulo en el componente', () => {
        
        render(
            <GifItem { ...spiderman }/>
        );
        
        expect( screen.getByText( spiderman.title ) ).toBeTruthy();

    });

});