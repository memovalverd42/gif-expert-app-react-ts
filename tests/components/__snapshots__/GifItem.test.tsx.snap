// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests del componente <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Spiderman"
      src="asdfadsfasdf"
    />
    <p>
      Spiderman
    </p>
  </div>
</div>
`;
